File: sion_fasion_shop/ProyectoMVC/SionFashioWebApplicationMVC/SionFashioWebApplicationMVC/Models/cliente.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace SionFashioWebApplicationMVC.Models;

public partial class cliente
{
    [Key]
    public int id_cliente { get; set; }

    [StringLength(255, ErrorMessage = "El nombre del cliente no puede exceder los 255 caracteres.")]
    public string? nombre_cliente { get; set; }

    [EmailAddress(ErrorMessage = "El correo electrónico no es válido.")]
    [StringLength(255, ErrorMessage = "El correo electrónico no puede exceder los 255 caracteres.")]
    public string? email_cliente { get; set; }

    [Phone(ErrorMessage = "El número de teléfono no es válido.")]
    [StringLength(255, ErrorMessage = "El número de teléfono no puede exceder los 255 caracteres.")]
    public string? telefono_cliente { get; set; }

    [StringLength(255, ErrorMessage = "La dirección no puede exceder los 255 caracteres.")]
    public string? direccion_cliente { get; set; }

    [InverseProperty("id_clienteNavigation")]
    public virtual ICollection<factura> facturas { get; set; } = new List<factura>();

    [InverseProperty("id_clienteNavigation")]
    public virtual ICollection<ordenes_de_compra> ordenes_de_compras { get; set; } = new List<ordenes_de_compra>();
}

File: sion_fasion_shop/Entregables/Trimestre V/Sistema de inventario en Node.Js/Bck-End SionFashion-Shop/models/cliente.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

const clienteSchema = new Schema({

    nombre_cliente: {
        type: String,
        maxlength: [255, 'El nombre del cliente no puede exceder los 255 caracteres.'],
        default: null
    },
    email_cliente: {
        type: String,
        maxlength: [255, 'El correo electrónico no puede exceder los 255 caracteres.'],
        match: [/.+@.+\..+/, 'El correo electrónico no es válido.'], // Validación de formato de email
        default: null
    },
    telefono_cliente: {
        type: String,
        maxlength: [255, 'El número de teléfono no puede exceder los 255 caracteres.'],
        match: [/^\+?[0-9\s]+$/, 'El número de teléfono no es válido.'], // Validación de formato de teléfono
        default: null
    },
    direccion_cliente: {
        type: String,
        maxlength: [255, 'La dirección no puede exceder los 255 caracteres.'],
        default: null
    },
    facturas: [{
        type: Schema.Types.ObjectId,
        ref: 'Factura' // Asegúrate de que este sea el nombre correcto del modelo factura
    }],
    ordenes_de_compras: [{
        type: Schema.Types.ObjectId,
        ref: 'OrdenesDeCompra' // Asegúrate de que este sea el nombre correcto del modelo orden de compra
    }]
});

// Exportar el modelo
module.exports = mongoose.model('Cliente', clienteSchema);
